src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -3884,22 +3884,11 @@ private static bool CanReuseParameter(CSharp.Syntax.ParameterSyntax parameter)
             this.ParseAttributeDeclarations(attributes);
             this.ParseParameterModifiers(modifiers);
 
-            var hasArgList = this.CurrentToken.Kind == SyntaxKind.ArgListKeyword;
-
             TypeSyntax type = null;
-            if (!hasArgList)
-            {
-                type = this.ParseType(mode: ParseTypeMode.Parameter);
-            }
-            else if (this.IsPossibleType())
-            {
-                type = this.ParseType(mode: ParseTypeMode.Parameter);
-                type = WithAdditionalDiagnostics(type, this.GetExpectedTokenError(SyntaxKind.CloseParenToken, SyntaxKind.IdentifierToken, 0, type.Width));
-            }
-
             SyntaxToken name = null;
```

Comment:
Consider setting `type = null;` explicitly in the `else` branch and removing the initializers for `type` and `name`.